### DISCUSSION TITLE
Designer for Blazor pages/components

### DISCUSSION THREAD

**Discussion by madev4cloud** [dotnet/aspnetcore#50521] [General] [answered]
I am new in Blazor. I was mostly devops and backend developer but I was Android dev 10 years ago. I remember using Eclipse or Android Studio there always was designer for UI item where You can prototype form look. I was easy to build base form then polish later in real device.

Now we have 2023 I am going to Visual Studio try to do Blazor but no designer, no preview, no prototype. I feel like child in mist. Every time I need to recompile project , this hot reload is taking time as well, not all items can be hot-reloaded. Should I see code and image hot it will look? Blazor development is so frustrating with Visual Studio.

Why no designer for Blazor development?

**Comment by marinasundstrom** [comment, marked as answer]
You should check out Radzen Blazor Studio: https://www.radzen.com/blazor-studio/

It has got what you want for Blazor.

The general trend in Web development has been moving away from WYSIWYG - and I think that it at least partly was because of the browser wars and not being compatible - before HTML5. Designers weren't able to handle the inconsistencies between browsers - or even catch up with new stuff. 

Today, most developers are now used to writing HTML and JavaScript, and preview in the browser as it has got all the tools necessary built-in. And there are tools that automatically reloads the browser - such as Hot Reload. WYSIWYG designers is not very suitable for building complex apps that have complex interactivity. You need to run them in the browser to test them.
